src/late_multicellular_stage/DualContourer.cs

```diff
@@ -0,0 +1,423 @@
+﻿using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Godot;
+using Godot.Collections;
+
+public class DualContourer
+{
+    public int PointsPerUnit = 4;
+
+    /// <summary>
+    ///   Specifies start of generated box. Inclusive.
+    /// </summary>
+    public Vector3 UnitsFrom = new(-4.0f, -4.0f, -4.0f);
+
+    /// <summary>
+    ///   Specifies end of generated box. Inclusive.
+    /// </summary>
+    public Vector3 UnitsTo = new(4.0f, 4.0f, 4.0f);
+
+    public bool Smoothen = true;
+
+    public IMeshGeneratingFunction? MathFunction;
```

Comment:
This seems like it should be a constructor parameter (and made non-nullable) as it doesn't seem like this class can do much if this is null.